src/ucp/rndv/rndv.c

```diff
@@ -406,11 +382,15 @@ static void ucp_rndv_req_send_rtr(ucp_request_t *rndv_req, ucp_request_t *rreq,
                                   ucs_ptr_map_key_t sender_req_id,
                                   size_t recv_length, size_t offset)
 {
+    ucp_ep_h ep = rndv_req->send.ep;
+
     ucp_trace_req(rndv_req, "send rtr remote sreq_id 0x%"PRIxPTR" rreq %p",
                   sender_req_id, rreq);
 
+    ucs_hlist_add_tail(&ucp_ep_ext_gen(ep)->proto_reqs, &rndv_req->list_elem);
```

Comment:
why not adding it inside ucp_proto_progress_rndv_rtr after succesful send, like in other places?

Reply:
done